src/proxy/inbound.rs

```diff
@@ -363,12 +363,26 @@ impl Inbound {
         upstream: &Workload,
         conn: &Connection,
     ) -> (bool, bool) {
-        let has_waypoint = !upstream.waypoint_addresses.is_empty();
-        for i in upstream.waypoint_addresses.iter() {
-            if workloads.fetch_workload(i).await.map(|w| w.identity()) == conn.src_identity {
-                return (has_waypoint, true);
+        let has_waypoint = upstream.waypoint.address.is_some();
+        let wp_ip = match upstream.waypoint.address.as_ref() {
+            Some(addr) => match addr {
+                gatewayaddress::Address::IP(wp_ip) => wp_ip,
+                gatewayaddress::Address::Hostname(_) => return (has_waypoint, false),
```

Comment:
can we add TODO to look this up from Service?

Reply:
see https://github.com/istio/ztunnel/pull/486/commits/f319ce5041d37875353d2b5acf685200caeb535b